### ISSUE TITLE
Cannot verify a contract with construct args and created within another contract

### ISSUE THREAD

**Issue by zkbenny** [matter-labs/hardhat-zksync#350]
I create a verify example [project](https://github.com/zkbenny/zksync-verify-contract-example) follow the guide [Verify Contracts with Hardhat](https://era.zksync.io/docs/dev/how-to/verify-contracts.html). Deploy and verify of the Greeter contract are ok, but verify a contract with construct args and created within another contract failed with message: 
```
Error in plugin @matterlabs/hardhat-zksync-verify: Constructor arguments are not correct
```

Here are two contracts: `DeployFactory` and `Proxy`, when deploy a `DeployFactory`, a `Proxy` with a default input `100` will be created within.

```solidity
// SPDX-License-Identifier: MIT OR Apache-2.0

pragma solidity ^0.8.0;

import "./Proxy.sol";

contract DeployFactory {
    Proxy public proxy;

    constructor() {
        proxy = new Proxy(100);
    }
}

```

```solidity
// SPDX-License-Identifier: MIT OR Apache-2.0

pragma solidity ^0.8.0;

contract Proxy {
    uint256 public count;

    constructor(uint256 _count) {
        count = _count;
    }
}

```

Here is the details of running deploy script to testnet:
```bash
yarn hardhat deploy-zksync --script deploy-proxy.ts

Running deploy script for the Proxy contract
The deployment is estimated to cost 0.000785932 ETH
Constructor args:0x
DeployFactory was deployed to 0x8c03eC86967347dEDB5f1614b0AB23e48F6a215D
Proxy was deployed to 0x22EA9D66eA20D0826444B9716Af4261Ce98b51Ba
Compiling contracts for zkSync Era with zksolc v1.3.13 and solc v0.8.17
Compiling 2 Solidity files
Your verification ID is: 34368
Contract successfully verified on zkSync block explorer!
Compiling contracts for zkSync Era with zksolc v1.3.13 and solc v0.8.17
Compiling 1 Solidity file
Your verification ID is: 34369
Error in plugin @matterlabs/hardhat-zksync-verify: Constructor arguments are not correct

For more info run Hardhat with --show-stack-traces
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The `DeployFactory` contract [0x8c03eC86967347dEDB5f1614b0AB23e48F6a215D](https://goerli.explorer.zksync.io/address/0x8c03eC86967347dEDB5f1614b0AB23e48F6a215D#contract) verify succeed, but the `Proxy` contract [0x22EA9D66eA20D0826444B9716Af4261Ce98b51Ba](https://goerli.explorer.zksync.io/address/0x22EA9D66eA20D0826444B9716Af4261Ce98b51Ba#contract) verify failed.

**Comment by zkbenny** [comment]
I'm sure the `constructorArguments` to verify `Proxy` contract is correct. I also tried verify command:
```bash
yarn hardhat verify --network zkSyncTestnet 0x22EA9D66eA20D0826444B9716Af4261Ce98b51Ba 100
```

Still not work.

```bash
Your verification ID is: 34388
Error in plugin @matterlabs/hardhat-zksync-verify: Constructor arguments are not correct
```

**Comment by zkbenny** [comment]
I also tried to verify in the explorer ui

<img width="1249" alt="image" src="https://github.com/matter-labs/hardhat-zksync/assets/85873552/de954d0e-9c04-438b-9bcc-512ae292a802">

but still not work
<img width="1267" alt="image" src="https://github.com/matter-labs/hardhat-zksync/assets/85873552/9adaa36e-7bea-4040-b4a0-2287faa7c4ee">

**Comment by zkbenny** [comment]
I tried looking for a solution from discord, and found a similar [question](https://discord.com/channels/722409280497516566/829104124145696808/1098715007941881946)

<img width="1087" alt="image" src="https://github.com/matter-labs/hardhat-zksync/assets/85873552/f9266f22-e2ad-4426-ac5e-c39a7af41d18">

**Comment by FilipTxFusion** [comment]
Hi @zkbenny, this is a known bug where the verification of the smart contract deployed from another contract expects parent's constructor args.
I will leave the issue open until the problem is resolved, and we will let you know once the new release is ready.

**Comment by zkbenny** [comment]
@FilipTxFusion Thanks, could you please fix this issue as soon as possible, if the contract cannot be open source and we cannot gain the trust of users

**Comment by statskiy-sovetnik** [comment]
@FilipTxFusion bumping this issue as we are also trying to verify a BeaconProxy, deployed from within a different contract. Getting an out-of-bound error, because I think it tries to decode tx params as constructor arguments, but they are different. 

![image](https://github.com/matter-labs/hardhat-zksync/assets/32808977/20fda2c5-7073-4fc0-a2e9-9c58a5673a20)

**Comment by FilipTxFusion** [comment]
I understand, thanks for reporting it. 
This task is added to our backlog and we will get to work on it as soon as other priorities are solved.

**Comment by novaknole** [comment]
Hi.. Is this issue resolved ? still getting the: `Backend verification error: Constructor arguments are not correct` when contract is deployed from another contract.

**Comment by zkcarter** [comment]
Hi @FilipTxFusion Does this problem solved? I see the official contract of [this](https://explorer.zksync.io/address/0x1F13810c6fFC29a9a26456f1D8541e4631178eaB#contract) has been verified.
I want to know how can our developer verify our contracts like zkSync L2 bridge token impl. Thank you and waiting for your reply.

**Comment by Jrigada** [comment]
Hey!
We've tested factory-deployed contract verification with multiple scenarios and is now working successfully on the latest version. If you're still experiencing verification issues with factory-deployed contracts or any other, please feel free to open a new issue with details about your specific case and we'll investigate further.
